asst1/CODE/my_scheduler.c

```diff
@@ -111,19 +111,40 @@ void run_maintenance() {
       if (new_prio == thread->priority) { // do nothing
         prev_ptr = ptr;
         ptr = ptr->next;
-      } else { // remove thread from current queue and add it to lower queue
+      } 
+      else { // remove thread from current queue and add it to lower queue
         thread->priority = new_prio;
         insert_ready_q(thread, new_prio);
         if (ptr == ready_q[i]->head) {
           ready_q[i]->head = ptr->next;
-        } else {
+        } 
+	else {
           prev_ptr->next = ptr->next;
         }
         if (ptr == ready_q[i]->tail) {
-          ready_q[i]->tail = NULL;
+          ready_q[i]->tail = prev_ptr;
         }
+	node_t* temp = ptr;
         ptr = ptr->next;
+	free(temp);
```

Comment:
why are we doing this?

Reply:
no garbage collection- even tho we can’t access the node anymore, need to free it’s malloc’d space

Reply:
but the object's still in use in another queue

Reply:
tcb* is in use but the node isn’t, just freeing the node not the thread block

Reply:
oh yes you're right